### PR TITLE
feat: validate edge function signature at build time

### PR DESCRIPTION
**Pull request by jackiewmacharia**
🎉 Thanks for sending this pull request! 🎉

Please make sure the title is clear and descriptive.

If you are fixing a typo or documentation, please skip these instructions.

Otherwise please fill in the sections below.

**Which problem is this pull request solving?**

Example: I'm always frustrated when [...]

**List other issues or pull requests related to this problem**

This fixes https://github.com/netlify/pod-compute/issues/249

**Describe the solution you've chosen**

Example: I've fixed this by [...]

**Describe alternatives you've considered**

Example: Another solution would be [...]

**Checklist**

Please add a `x` inside each checkbox:

- [ ] I have read the [contribution guidelines](../blob/main/CONTRIBUTING.md).
- [ ] The status checks are successful (continuous integration). Those can be seen below.

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: deno/config.ts [hunk 1]
```diff
@@ -11,6 +11,14 @@ try {
   Deno.exit(exitCodes.ImportError)
 }
 
+if (func.default === undefined) {
```
Comment (eduardoboucas) [inline deno/config.ts:14]: I understand that this level of granularity gives people more information — i.e. no export vs. export with wrong type. But do you think it's worth the additional logic for the extra branch? If we just do the `typeof` check and show the same message for the two cases with a `ntl.fyi` link pointing to https://docs.netlify.com/edge-functions/api/, I think that would achieve the same goal with half the logic.

What do you think?

Comment (jackiewmacharia) [reply deno/config.ts:14]: I think it makes sense, I'll implement it.

File: node/config.ts [hunk 2]
```diff
@@ -121,6 +124,12 @@ const logConfigError = (func: EdgeFunction, exitCode: number, stderr: string, lo
 
       break
 
+    case ConfigExitCode.InvalidDefaultExport:
+      errMsg = `Default export in edge function at '${func.path}' must be a function. More on the Edge Functions API at https://docs.netlify.com/edge-functions/api/`
```
Comment (jackiewmacharia) [inline node/config.ts:128]: This needs a `ntl.fyi` shortened link

File: deno/config.ts (hunk 1 above)
Comment (jackiewmacharia) [reply deno/config.ts]: This is now done _(waiting on a `ntl.fyi` link though)_.

File: node/config.ts [hunk 3]
```diff
@@ -18,6 +18,8 @@ enum ConfigExitCode {
   InvalidExport,
   RuntimeError,
   SerializationError,
+  NoDefaultExport,
```
Comment (eduardoboucas) [inline node/config.ts:21]: I think this is not needed anymore?

File: node/config.ts [hunk 4]
```diff
@@ -121,6 +124,12 @@ const logConfigError = (func: EdgeFunction, exitCode: number, stderr: string, lo
 
       break
 
+    case ConfigExitCode.InvalidDefaultExport:
+      errMsg = `Default export in edge function at '${func.path}' must be a function. More on the Edge Functions API at https://ntl.fyi/edge-api`
```
Comment (eduardoboucas) [inline node/config.ts:128]: Do we need to initialise this variable at the top and mutate it here? Could we instead just send the string directly to the `Error` constructor?

File: node/config.ts [hunk 5]
```diff
@@ -121,6 +124,12 @@ const logConfigError = (func: EdgeFunction, exitCode: number, stderr: string, lo
...
+      throw new Error(errMsg)
+
+      break
```
Comment (eduardoboucas) [inline node/config.ts:131]: This is unreachable code since we're throwing above.

File: node/config.test.ts [hunk 6]
```diff
@@ -21,6 +21,9 @@ const importMapFile = {
   },
 }
 
+const invalidDefaultExportErr = (path: string) =>
+  `Default export in edge function at '${path}' must be a function. More on the Edge Functions API at https://ntl.fyi/edge-api`
```
Comment (eduardoboucas) [inline node/config.test.ts:25]: Small nit that you're free to ignore, but I feel that we could make the message shorter by not repeating "edge function".

```suggestion
  `Default export in '${path}' must be a function. More on the Edge Functions API at https://ntl.fyi/edge-api.`
```

File: deno/config.ts (hunk 1 above)
Comment (jackiewmacharia) [reply deno/config.ts]: now done done

**Review by eduardoboucas — COMMENTED** [review]
I left a few small comments, but otherwise looking great!

**Review by eduardoboucas — APPROVED** [review]
🚀

**Pull request merged**